corgidrp/l2a_to_l2b.py

```diff
@@ -58,6 +58,38 @@ def dark_subtraction(input_dataset, dark_frame):
 
     return darksub_dataset
 
+def flat_division(input_dataset, master_flat):
+    """
+    
+    Divide the dataset by the master flat field. 
+
+    Args:
+        input_dataset (corgidrp.data.Dataset): a dataset of Images (L2a-level)
+        master_flat (corgidrp.data.Flat): a master flat field to divide by
+
+    Returns:
+        corgidrp.data.Dataset: a version of the input dataset with the flat field divided out
+    """
+    
+     # copy of the dataset
+    flatcal_dataset = input_dataset.copy()
+    
+    #Divide by the master flat
+    flatcal_cube = flatcal_dataset.all_data/master_flat.data
+    
+    # propagate the error of the master flat frame # Check this error prop 
+    if hasattr(master_flat, "err"):
+        flatcal_dataset.add_error_term(master_flat.err, "masterflat_error")   
```

Comment:
Also: 
Here the error in the test is happening because the dataset.add_error_term is expecting a 3D cube of shape: [n_frames,xdim,ydim]
or a 2D frame of shape [xdim,ydim]. Since we're trying to keep ALL of our image.err formats as 3D cubes [n_layer,xdim,ydim] (and we have hidden functions that try to enforce this when reading in data), I think you're inadvertantly passing in a 3D cube here.